### DISCUSSION TITLE
Auto Merge using Python

### DISCUSSION THREAD

**Discussion by rohnnie** [community/community#32187] [Pull Requests]
I have written a python script that creates a branch and raises a Pull Request. Now I need to write code for merging that branch into master after all checks have passed. Is there any way that I can achieve this using python?

**Comment by xinxinxiangying258** [comment]
I remember that in the setting page of repository has options that auto-merge a pull request.

**Reply by rohnnie** [reply]
I need to implement it using python. Could you suggest how could we achieve this?

**Reply by airtower-luna** [reply]
That setting has to be enabled *before* auto-merge can be used. It doesn't enable auto-merge for a specific PR.

**Comment by airtower-luna** [comment]
You'll have to use the API, which you can do with any HTTP client module. [`requests`](https://requests.readthedocs.io/en/latest/) is what I'd use. You wrote that your script already creates a PR, so I assume you already use the API.

If you want to check what checks ran and their results, you can use [the checks API](https://docs.github.com/en/rest/checks/suites#list-check-suites-for-a-git-reference) and [do the merge](https://docs.github.com/en/rest/pulls/pulls#merge-a-pull-request) if the ones you want passed.

If you want to enable auto-merge (where GitHub does the merge as soon as the checks required by a branch protection rule pass) you'll need to [use the GraphQL API](https://docs.github.com/en/graphql/reference/mutations#enablepullrequestautomerge), because there doesn't seem to be a REST endpoint to enable auto-merge.

Either way it's HTTPS, and your client needs to provide an `Authorization` header. :slightly_smiling_face:
